stock_picking_group_by_partner_by_carrier/models/stock_move.py

```diff
@@ -53,59 +53,52 @@ def _on_assign_picking_message_link(self):
                     subtype_id=self.env.ref("mail.mt_note").id,
                 )
 
-    def _search_picking_for_assignation(self):
-        # totally reimplement this one to add a hook to change the domain
-        self.ensure_one()
-        picking = self.env["stock.picking"].search(
-            self._domain_search_picking_for_assignation(), limit=1
-        )
-        return picking
-
-    def _domain_search_picking_handle_move_type(self):
-        """Hook to handle the move type. Can be overloaded by other modules.
-        By default the move type is taken from the procurement group.
-        """
-        # avoid mixing picking policies
-        return [("move_type", "=", self.group_id.move_type)]
-
     def _domain_search_picking_for_assignation(self):
-        states = ("draft", "confirmed", "waiting", "partially_available", "assigned")
+        domain = super()._domain_search_picking_for_assignation()
         if (
             not self.picking_type_id.group_pickings
             or self.group_id.sale_id.picking_policy == "one"
         ):
-            # use the normal domain from the stock module
-            domain = [
-                ("group_id", "=", self.group_id.id),
+            return domain
```

Comment:
@simahawk Should we also call `grouping_domain = self._assign_picking_group_domain()` and add it to `domain` before returning?

Reply:
Hmmm this is actually returning the default domain. It was already like this before the split.

Reply:
Before we were not returning `domain`, and we were reaching the following block code:
```python
        if self.env.context.get("picking_no_copy_if_can_group"):
            # we are in the context of the creation of a backorder:
            # don't consider the current move's picking
            domain.append(("id", "!=", self.picking_id.id))
```
And this filter on `id` was enough to not get Shopfloor test failing. So the behavior has changed.

Reply:
FTR we said that the fix should be in `_create_backorder` which should not set the flag if picking type is not grouping enabled

Reply:
Fixed here: https://github.com/OCA/stock-logistics-workflow/pull/757